src/idl_parser.cpp

```diff
@@ -2984,6 +3013,11 @@ CheckedError Parser::ParseProtoFields(StructDef *struct_def, bool isextend,
       }
       if (!field) ECHECK(AddField(*struct_def, name, type, &field));
       field->doc_comment = field_comment;
+      if (!id.empty() || oneof) {
+        auto val = new Value();
```

Comment:
Is this the way we typically make `Value`? who deletes this memory?

Reply:
Yes, I prefer use unique_ptr, but it needs a lot of changes, 
It will be deleted in [SymbolTable destructor](https://github.com/google/flatbuffers/blob/e1a2f688e0d12385fd977cb45d58453d88d7c0e2/include/flatbuffers/idl.h#L203).